support/globalconfig/dns.go

```diff
@@ -9,26 +9,44 @@ import (
 	hyperv1 "github.com/openshift/hypershift/api/v1alpha1"
 )
 
-func DNSConfig() *configv1.DNS {
-	return &configv1.DNS{
+func DNSConfig(hcp *hyperv1.HostedControlPlane) *configv1.DNS {
+	dnsConfig := &configv1.DNS{
 		ObjectMeta: metav1.ObjectMeta{
 			Name: "cluster",
 		},
 	}
-}
-
-func ReconcileDNSConfig(dns *configv1.DNS, hcp *hyperv1.HostedControlPlane) {
-	dns.Spec.BaseDomain = BaseDomain(hcp)
+	if hcp.Spec.Platform.Type == hyperv1.IBMCloudPlatform {
```

Comment:
This change can go in the ReconcileDNSConfig function

Reply:
It can as long as we always run reconcileDNS

Reply:
I was thinking "reconcile" is really like a kube operation if that makes sense not just simply getting a desired config? It seemed weird that we were calling reconcile in "param functions" but I'm happy for anything.

Reply:
will refactor